components/supervisor/pkg/supervisor/tasks.go

```diff
@@ -377,9 +379,29 @@ func (tm *tasksManager) watch(task *task, terminal *terminal.Term) {
 	go func() {
 		defer stdout.Close()
 
-		fileName := tm.prebuildLogFileName(task)
-		// TODO(janx): If the file already exists (from a parent prebuild), extract its "time saved", and log that below
-		// (instead, or in addition to, the incremental prebuild time).
+		var (
+			fileName                    = tm.prebuildLogFileName(task)
+			parentLog                   = ""
+			parentElapsed time.Duration = 0
+		)
+		if _, err := os.Stat(fileName); err == nil {
+			// If the file already exists (from a parent prebuild), extract its "elapsed time" before overwriting the prebuild logs.
+			content, err := ioutil.ReadFile(fileName)
+			if err == nil {
```

Comment:
The block below is nested unusually deep. I'd pull the entire thing out into a function that reads the file (using a scanner) and returns the time the previous prebuild took. Something like:

```Go
func getParentBuildDuration(fn string) time.Duration {
    if _, err := os.Stat(fn); err != nil {
        return 0
    }

    file, err := os.Open(fn)
    if err != nil {
        // can't open the file - touch luck
       return 0
    }
    defer file.Close()

    scanner := bufio.NewScanner(file)
    for scanner.Scan() {
        l := scanner.Text()
        if !reg.Match(l) {
            continue
        }
        res := time.Duration(extractDurationUsingRegexp(reg, l))
       return res
    }

    return 0
}
```